homeassistant/helpers/entity_registry.py

```diff
@@ -172,32 +172,35 @@ def write_unavailable_state(self, hass: HomeAssistant) -> None:
         hass.states.async_set(self.entity_id, STATE_UNAVAILABLE, attrs)
 
 
-class EntityRegistryStore(storage.Store):
+class EntityRegistryStore(storage.Store[dict[str, list[dict[str, Any]]]]):
     """Store entity registry data."""
 
     async def _async_migrate_func(
-        self, old_major_version: int, old_minor_version: int, old_data: dict
+        self,
+        old_major_version: int,
+        old_minor_version: int,
+        old_data: dict[str, list[dict[str, Any]]],
     ) -> dict:
         """Migrate to the new version."""
         data = old_data
         if old_major_version == 1 and old_minor_version < 2:
             # From version 1.1
             for entity in data["entities"]:
                 # Populate all keys
-                entity["area_id"] = entity.get("area_id")
-                entity["capabilities"] = entity.get("capabilities") or {}
-                entity["config_entry_id"] = entity.get("config_entry_id")
-                entity["device_class"] = entity.get("device_class")
-                entity["device_id"] = entity.get("device_id")
-                entity["disabled_by"] = entity.get("disabled_by")
-                entity["entity_category"] = entity.get("entity_category")
-                entity["icon"] = entity.get("icon")
-                entity["name"] = entity.get("name")
-                entity["original_icon"] = entity.get("original_icon")
-                entity["original_name"] = entity.get("original_name")
-                entity["platform"] = entity["platform"]
-                entity["supported_features"] = entity.get("supported_features", 0)
-                entity["unit_of_measurement"] = entity.get("unit_of_measurement")
+                entity.setdefault("area_id", None)
+                entity.setdefault("capabilities", {})
+                entity.setdefault("config_entry_id", None)
+                entity.setdefault("device_class", None)
+                entity.setdefault("device_id", None)
+                entity.setdefault("disabled_by", None)
+                entity.setdefault("entity_category", None)
+                entity.setdefault("icon", None)
+                entity.setdefault("name", None)
+                entity.setdefault("original_icon", None)
+                entity.setdefault("original_name", None)
+                entity.setdefault("platform", None)
```

Comment:
We didn't have a default for platform in the old code.

Reply:
Why don't we need platform now?

Reply:
The original code just did this, which was clearly meaningless:
```py
entity["platform"] = entity["platform"]
```

Reply:
Right, but it was not wrong in another way, and still needed? When was platform introduced and why was it there in the first place?